### DISCUSSION TITLE
[React][tailwindcss] double style not use the last style

### DISCUSSION THREAD

**Discussion by robyhuzwandar** [tailwindlabs/tailwindcss#9835] [Help]
What's wrong if I have component text with the default `font-weight:400` and then I send a new `font-weight via props` with `font-weight:600` but the component still uses `font-weight:400` and crosses out my `font-weight:600`?

**Inspect Element:**
<img width="331" alt="Screen Shot 2022-11-14 at 18 07 09" src="https://user-images.githubusercontent.com/20414047/201777487-559b1159-9b1a-4060-9bff-25a710b4d260.png">

but when I try on the [tailwind playground](https://play.tailwindcss.com/) it should use the last style and ignore the old one

**My expectation:**
<img width="1428" alt="Screen Shot 2022-11-15 at 05 47 03" src="https://user-images.githubusercontent.com/20414047/201777517-ad004406-44b5-416e-967c-2bbaddb0e813.png">

is there any solution without use of `!` on each style

**My Component:**
![image](https://user-images.githubusercontent.com/20414047/201777756-cb7341da-edbe-4014-b6b5-14c02426b1cc.png)

**Comment by wongjn** [comment]
You should structure your code such that only one class name that applies the same property value in the same context is applied on an element at any one time. So for your case, consider reworking the class names logic so that only `font-semibold` or `font-normal` appear in the class names list, but not at the same time.

Your issue arises from the fact that for two rules with the same selector specificity, the rule that appears later in the stylesheet will take precedence over the former.
